hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordSource.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.records;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.TransactionID;
+import com.hedera.hapi.node.transaction.TransactionReceipt;
+import com.hedera.hapi.node.transaction.TransactionRecord;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * A source of {@link TransactionRecord}s and {@link TransactionReceipt}s for one or more {@link TransactionID}'s.
```

Comment:
I see this maintains more than one `TransactionID` records. From the recording I see its only because of `Upgrade` system file updates uses 0.0.50 as payer, which will change the TransactionID. We should probably add javadoc regarding this.

Reply:
Also a `FUTURE` that this will maintain only one `TransactionID` if we make SystemFileUpdates are dispatched use separate stack.

Reply:
Done.